mhs/mhs/common/state/work_description.py

```diff
@@ -0,0 +1,155 @@
+import json
+import utilities.integration_adaptors_logger as log
+import datetime
+
+logger = log.IntegrationAdaptorsLogger('STATE_MANAGER')
+
+
+class MessageStatus:
+    RECEIVED = 1
+    STARTED = 2
+    IN_OUTBOUND_WORKFLOW = 3
+
+
+DATA_KEY = 'MESSAGE_KEY'
+VERSION_KEY = 'VERSION'
+CREATED_TIMESTAMP = 'CREATED'
+LAST_MODIFIED_TIMESTAMP = 'TIMESTAMP'
+DATA = 'DATA'
+STATUS = 'STATUS'
+
+
+class OutOfDateVersionError(RuntimeError):
+    """Exception thrown when trying to update a state and the local version is behind the remote version"""
+    pass
+
+
+class EmptyWorkDescriptionError(RuntimeError):
+    """Exception thrown when no work description is found for a given key"""
+    pass
+
+
+def get_time():
+    """Returns UTC time in the appropriate format """
+    return datetime.datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%f')
+
+
+async def get_work_description_from_store(persistence_store, key: str):
```

Comment:
```suggestion
async def get_work_description_from_store(persistence_store, key: str) -> WorkDescription:
```
Might have to do that as a string or do `from __future__ import annotations`